### PR TITLE
INT-3991: Add `IntegrationGraphController`

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-3991
- Add `IntegrationGraphController` `@RestController` over `IntegrationGraphServer` bean
- Add `@EnableIntegrationGraphController` and `<int-http:graph-controller>` to register
  `IntegrationGraphController` if the `DispatcherServlet` is in classpath.
- Also register the `IntegrationGraphServer` bean from the same place, if that isn't presented in the application context yet.
- Allow to configure the "root" `path` for the `IntegrationGraphController` as a placeholder value via
  `spring.integration.graph.controller.request.mapping.path` property
- Add tests for both `@EnableIntegrationGraphController` and `<int-http:graph-controller>` cases based on the `MockMvc`

### CODE REVIEW

**Comment by artembilan** [comment]
Need Docs and maybe some JavaDocs.
But, at least, it will be a chance for you to review already.

**Comment by garyrussell** [comment]
Cool - I think we can leave it undocumented in M2.

File: spring-integration-http/src/main/java/org/springframework/integration/http/support/HttpContextUtils.java [hunk 1]
```diff
@@ -30,8 +33,33 @@ private HttpContextUtils() {
 	}
 
 	/**
+	 * The {@code boolean} flag to indicate if the {@code org.springframework.web.servlet.DispatcherServlet}
+	 * is presented in the CLASSPATH to allow to register the Integration server components,
```
Comment (garyrussell) [inline spring-integration-http/src/main/java/org/springframework/integration/http/support/HttpContextUtils.java]: `s/presented/present/` (x2) will fix on merge.

**Comment by garyrussell** [comment]
Merged as 8df487c96ffadf125a6911c6bbf2771f14f26746

**Pull request closed without merging**